File: backend/app/db/supabase.py
from supabase import Client, create_client

from app.core.config import settings


def create_admin_client() -> Client:
    """secret 키를 쓰는 클라이언트. RLS를 우회한다.

    사용자와 무관한 배치 작업(RSS 수집, 콘텐츠 저장)에만 쓴다.
    사용자 데이터를 다룰 때는 create_user_client를 쓴다.
    """
    return create_client(settings.supabase_url, settings.supabase_secret_key)


def create_user_client(access_token: str) -> Client:
    """사용자의 Supabase 세션 토큰으로 요청하는 클라이언트.

    publishable 키 + 사용자 JWT를 함께 보내므로 RLS가 그대로 적용된다.
    사용자 데이터(user_interests, mission_records)는 이 클라이언트로 접근한다.
    """
    client = create_client(settings.supabase_url, settings.supabase_publishable_key)
    client.postgrest.auth(access_token)
    return client

File: server/src/config/supabase.js
import { createClient } from "@supabase/supabase-js";

// service_role(비밀) 키 — 브라우저는 Supabase에 직접 붙지 않고 항상 이 서버(Express)를 거치므로,
// RLS를 우회하는 service_role을 서버 전용으로 쓴다. anon 키는 브라우저 직접 연결용이라 여기선 안 씀.
const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;

export const supabase = supabaseUrl && supabaseKey
  ? createClient(supabaseUrl, supabaseKey)
  : null;
